app/preferences_ios.go

```diff
@@ -0,0 +1,126 @@
+// +build ios
+
+package app
+
+import (
+	"unsafe"
+
+	"fyne.io/fyne"
+)
+
+/*
+#import <stdlib.h>
+#import <Foundation/Foundation.h>
+
+int getPreferenceBool(const char* key, int fallback);
+void setPreferenceBool(const char* key, int value);
+float getPreferenceFloat(const char* key, float fallback);
+void setPreferenceFloat(const char* key, float value);
+int getPreferenceInt(const char* key, int fallback);
+void setPreferenceInt(const char* key, int value);
+const char* getPreferenceString(const char* key, const char* fallback);
+void setPreferenceString(const char* key, const char* value);
+*/
+import "C"
+
+type iOSPreferences struct {
+}
+
+// Declare conformity with Preferences interface
+var _ fyne.Preferences = (*iOSPreferences)(nil)
+
+func (p *iOSPreferences) Bool(key string) bool {
+	return p.BoolWithFallback(key, false)
+}
+
+func (p *iOSPreferences) BoolWithFallback(key string, fallback bool) bool {
+	cKey := C.CString(key)
+	cBool := 0
+	if fallback {
+		cBool = 1
+	}
+	ret := C.getPreferenceBool(cKey, C.int(cBool))
+
+	C.free(unsafe.Pointer(cKey))
+	return int(ret) != 0
+}
+
+func (p *iOSPreferences) SetBool(key string, value bool) {
+	cKey := C.CString(key)
+	cBool := 0
+	if value {
+		cBool = 1
+	}
+	C.setPreferenceBool(cKey, C.int(cBool))
+
+	C.free(unsafe.Pointer(cKey))
+}
```

Comment:
I propose to use `C.bool` for the bool values. You might need to import AppKit.h on Obj-C side.

Reply:
Sorry, I tried but objC.BOOL is not C.bool, so it still requires additional conversion.
I will give it another shot... Is the usage of “stdbool.h” truly universal for all CGo targets?
A cursory glance indicates that MS only made it available in 2013

Reply:
Who cares about M$ ;). This is MacOS code anyhow.
Maybe `BOOL` and `bool` are interoperable … give it a try.
If not, I still would prefer to have the conversion in the Obj-C code and use `bool` for Go<->ObjC communication.

Reply:
Haha, yes this is just for mac aspects - I forgot! Coding in 3 different languages concurrently seems to take it's toll :(

Reply:
I think that BOOL is "signed char" whereas bool is an "int", will need some conversion.

Reply:
I still think it's interoperable. The compiler probably will do the correct thing.